src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Android.Accounts;
+using Android.App;
+using Android.Content;
+using AndroidNative = Android;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal.Broker;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.UI;
+using Microsoft.Identity.Client.Utils;
+using System.Globalization;
+using Microsoft.Identity.Json.Linq;
+using Microsoft.Identity.Json;
+using Android.OS;
+
+namespace Microsoft.Identity.Client.Platforms.Android
+{
+    [AndroidNative.Runtime.Preserve(AllMembers = true)]
+    internal class AndroidBroker : IBroker
+    {
+        //Since the response for the broker interactive call is returned in the AuthenticationContinuationHelper, this class needs to send the response to 
+        //the broker and wait for the AuthenticationContinuationHelper to return the response to the broker thread. This sephamore is used to make the MSAL code wait
+        //for the response. The AuthenticationContinuationHelper will trigger the continueation of the broker thread once it sets the rsultEX value. 
+        //Because of this, resultEX is made to be static. However, there will only ever be one broker authentication happening at once.
+        private static SemaphoreSlim _readyForResponse = null;
+        private static MsalTokenResponse _androidBrokerTokenResponse = null;
+        //Since the correlation ID is not returned from the broker response, it must be stored at the beginning of the authentication call and reinjected into the response at the end.
+        private static string _correlationId;
+        private readonly AndroidBrokerHelper _brokerHelper;
+        private readonly ICoreLogger _logger;
+        private Activity _activity;
+
+
+        public AndroidBroker(ICoreLogger logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _brokerHelper = new AndroidBrokerHelper(Application.Context, logger);
+        }
+
+        public bool CanInvokeBroker(CoreUIParent uiParent)
+        {
+            bool canInvoke = _brokerHelper.CanSwitchToBroker();
+            _logger.Verbose("Can invoke broker? " + canInvoke);
+
+            _activity = uiParent?.Activity;
```

Comment:
This is an anti-pattern. Have a look at Martin Fowler's article "Command Query Separation" https://martinfowler.com/bliki/CommandQuerySeparation.html. 

Basically `CanInvokeBroker` is a query by name, but it acts as a command, as it sets the activity. Ideally the _activity is readonly and you get a hold of it in the ctor.